### ISSUE TITLE
Connect: `securityContext` for jobs (azure gatekeeper)

### ISSUE THREAD

**Issue by pat-s** [rstudio/helm#319]
A follow-up on #309 and #293.

With 0.3.11 and the following config

```yml
launcher:
  templateValues:
    pod:
      securityContext:
        runAsUser: 999
        runAsGroup: 999
        allowPrivilegeEscalation: false
    job:
      securityContext:
        runAsUser: 999
        runAsGroup: 999
        allowPrivilegeEscalation: false
```

I still get

```
Warning  FailedCreate  6s (x2 over 16s)  job-controller  Error creating: admission webhook "validation.gatekeeper.sh" denied the request: [azurepolicy-k8sazurev3noprivilegeescalatio-da2c940f21133e02a3d1] Privilege escalation container is not allowed: init
[azurepolicy-k8sazurev3noprivilegeescalatio-da2c940f21133e02a3d1] Privilege escalation container is not allowed: rs-launcher-container
```

Inspecting the blocked `job.batch/packrat-restore-*` manifests, I see that the `rs-launcher-container` has no `securityContext` applied. There is one for the `init` container but apparently this is also not yet enough to make the gatekeeper happy 🤔 

What makes this hard to debug is that every time I am trying to edit the job at hand, it complains about an invalid config. I cannot even add `allowPrivilegeEscalation: false` to existing `securityContext` blocks 🤔 

The above config does not seem to have any influence for the `job.batch/packrat-restore` being spawned when inspecting the resulting manifests. I've tried reading https://github.com/rstudio/helm/blob/main/charts/rstudio-connect/files/job.tpl but was not finally able to infer which settings from `values.yml` would make a change there.

Is there something I am overlooking?

**Comment by pat-s** [comment]
Good news first! I meanwhile got a whitelist for these policies on the client cluster! This means the pods are not blocked anymore but also that I don't have an environment anymore to test your fixes against 🥲 :/

But the definitions for `initContainer` were needed anyhow I think so these additions will definitely become valuable! I think one should be able to reproduce this and even test on a dev cluster on Azure but I don't have one right now (we're still early in our Azure playground attempts). 

Even though I don't know if this issue is now fixed with #322, feel free to close since I cannot actively test against this right now :)

**Comment by colearendt** [comment]
Thanks for the update!! Glad to hear the customer is unblocked, at least 😄 I'm hopeful #322 will resolve the issue! If not and this resurfaces, please feel free to reopen or open a related issue 😄 We need to level up our Azure test environments too
